### PR TITLE
Multisig command export and execute options

### PR DESCRIPTION
**Pull request by RodrigoAD**
- By default, it will prompt the tx execution data
- if flag `--execute` is provided, will use local wallet to sign and broadcast the tx

### CODE REVIEW

File: gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts [hunk 1]
```diff
@@ -132,71 +144,114 @@ export const wrapCommand = (command) => {
...
-          remainingAccounts: context.proposalState.accounts
-            .map((t) => (t.pubkey.equals(context.multisigSigner) ? { ...t, isSigner: false } : t))
-            .concat({
-              pubkey: context.proposalState.programId,
-              isWritable: false,
-              isSigner: false,
-            }),
+    executeProposal: ProposalAction = async (proposal: PublicKey, context): Promise<RawTransaction[]> => {
+      logger.loading(`Generating proposal execution data for ${command.id}`)
+      const data = this.program.coder.instruction.encode('executeTransaction', {})
```
Comment (archseer) [inline gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts:215]: Why the change here?

Comment (RodrigoAD) [reply gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts:218]: We don't want to execute the proposal directly, but generate the tx and decide based on the user instructions

File: gauntlet/packages/gauntlet-serum-multisig/src/commands/inspect.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { Result } from '@chainlink/gauntlet-core'
+import { logger } from '@chainlink/gauntlet-core/dist/utils'
+import { SolanaCommand, TransactionResponse } from '@chainlink/gauntlet-solana'
+import { PublicKey } from '@solana/web3.js'
+import { CONTRACT_LIST, getContract } from '@chainlink/gauntlet-solana-contracts'
+
+export default class MultisigCreate extends SolanaCommand {
```
Comment (krebernisak) [inline gauntlet/packages/gauntlet-serum-multisig/src/commands/inspect.ts:7]:
```suggestion
export default class MultisigInspect extends SolanaCommand {
```

File: gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts [hunk 3]
```diff
@@ -42,12 +41,52 @@ export const wrapCommand = (command) => {
...
+      const proposal = new PublicKey(this.flags.proposal || rawTxs[0].accounts[1].pubkey)
+      const latestSlot = await this.provider.connection.getSlot()
+      const recentBlock = await this.provider.connection.getBlock(latestSlot)
+      const tx = makeTx(rawTxs, {
+        recentBlockhash: recentBlock.blockhash,
+        feePayer: signer,
+      })
+      if (this.flags.execute) {
+        logger.loading(`Executing action...`)
+        const txhash = await this.provider.send(tx, [this.wallet.payer])
```
Comment (krebernisak) [inline gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts:58]: This will fail if the `this.flags.signer` is provided with `--execute`, right?

File: gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts [hunk 4]
```diff
@@ -42,12 +41,52 @@ export const wrapCommand = (command) => {
...
+              tx: this.wrapResponse(txhash, this.multisigAddress.toString()),
+              contract: this.multisigAddress.toString(),
+            },
+          ],
+        }
+      }
+
+      logger.log(`Transaction generated:
+        ${tx.compileMessage().serialize().toString('base64')}
+      `)
```
Comment (krebernisak) [inline gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts:72]: Expand this to two lines for readability :)

File: gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts (hunk 3 above)
Comment (RodrigoAD) [reply gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts:52]: Yeah true. If execute is provided `signer` should fall into `wallet.payer` by default

File: gauntlet/packages/gauntlet-solana-contracts/artifacts/schemas/serum_multisig.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "0.0.0",
+  "version": "0.8.0",
```
Comment (krebernisak) [inline gauntlet/packages/gauntlet-solana-contracts/artifacts/schemas/serum_multisig.json:2]: This one should be `"version": "0.9.0"`. You can take it from the multisig PR #157.

File: gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts (hunk 3 above)
Comment (krebernisak) [reply gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts:52]: I'm saying we shouldn't let it fail so late, but validate input earlier and if both are set then fail with the explanation why this configuration is not possible.

Comment (RodrigoAD) [reply gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts:54]: That make more sense. Changing it

File: gauntlet/packages/gauntlet-serum-multisig/src/commands/setOwners.ts [hunk 6]
```diff
@@ -2,14 +2,13 @@ import { SolanaCommand, RawTransaction, TransactionResponse } from '@chainlink/g
 import { PublicKey } from '@solana/web3.js'
 import { CONTRACT_LIST, getContract } from '@chainlink/gauntlet-solana-contracts'
 import { Result } from '@chainlink/gauntlet-core'
+import { logger } from '@chainlink/gauntlet-core/dist/utils'
 
 export default class SetOwners extends SolanaCommand {
-  static id = 'set:owners'
+  static id = 'set_owners'
```
Comment (krebernisak) [inline gauntlet/packages/gauntlet-serum-multisig/src/commands/setOwners.ts:8]: As we namespace other commands we should do the same here:

```suggestion
  static id = 'multisig:set_owners'
```

This is consistent with `multisig:inspect`.

File: gauntlet/packages/gauntlet-serum-multisig/src/commands/setOwners.ts [hunk 7]
```diff
@@ -2,14 +2,13 @@ import { SolanaCommand, RawTransaction, TransactionResponse } from '@chainlink/g
...
   static category = CONTRACT_LIST.MULTISIG
 
-  static examples = [
-    'yarn gauntlet-serum-multisig set:owners --network=local --approve --tx=9Vck9Gdk8o9WhxT8bgNcfJ5gbvFBN1zPuXpf8yu8o2aq --execute AGnZeMWkdyXBiLDG2DnwuyGSviAbCGJXyk4VhvP9Y51M QMaHW2Fpyet4ZVf7jgrGB6iirZLjwZUjN9vPKcpQrHs',
-  ]
+  static examples = ['yarn gauntlet-serum-multisig set_owners --network=local']
```
Comment (krebernisak) [inline gauntlet/packages/gauntlet-serum-multisig/src/commands/setOwners.ts:11]:
```suggestion
  static examples = ['yarn gauntlet-serum-multisig set_owners --network=local [OWNERS...]']
```

File: gauntlet/packages/gauntlet-serum-multisig/src/commands/setThreshold.ts [hunk 8]
```diff
@@ -2,15 +2,13 @@ import { SolanaCommand, RawTransaction, TransactionResponse } from '@chainlink/g
 import { PublicKey } from '@solana/web3.js'
 import { CONTRACT_LIST, getContract } from '@chainlink/gauntlet-solana-contracts'
 import { Result } from '@chainlink/gauntlet-core'
-import { BN } from '@chainlink/gauntlet-core/dist/utils'
+import { BN, logger } from '@chainlink/gauntlet-core/dist/utils'
 
 export default class SetThreshold extends SolanaCommand {
-  static id = 'set:threshold'
+  static id = 'set_threshold'
```
Comment (krebernisak) [inline gauntlet/packages/gauntlet-serum-multisig/src/commands/setThreshold.ts:8]:
```suggestion
  static id = 'multisig:set_threshold'
```

Comment (krebernisak) [reply gauntlet/packages/gauntlet-serum-multisig/src/commands/setThreshold.ts:8]: It should probably be `multisig:change_threshold` as this is the instruction we call.

**Review by krebernisak — APPROVED** [review]
